lldb/source/Plugins/SymbolLocator/Debuginfod/SymbolLocatorDebuginfod.cpp

```diff
@@ -133,10 +213,32 @@ static std::optional<FileSpec> GetFileForModule(
 
 std::optional<ModuleSpec> SymbolLocatorDebuginfod::LocateExecutableObjectFile(
     const ModuleSpec &module_spec) {
```

Comment:
This callback method would need to take an argument that specifies if this is user initiated like:
```
std::optional<ModuleSpec> SymbolLocatorDebuginfod::LocateExecutableObjectFile(
    const ModuleSpec &module_spec, bool user_initiated)
```
Then we can properly handle the `eLookupModeOnDemand` case. So we need to modify the static function in SymbolLocator to take that extra parameter and have it pass it along to this funciton.

Reply:
force_lookup already serves this purpose :)